### ISSUE TITLE
Ledger signMessage support

### ISSUE THREAD

**Issue by hudymvol** [solana-labs/wallet-adapter#171]
Hi, are there any plans to add `signMessage` support for Ledger? 

As far as I can see, there isn't at the current moment.

**Comment by neverything** [comment]
@hudymvol the issue seems to go a little deeper: https://github.com/phantom-labs/sandbox/issues/14

**Comment by volodymyrhudyma** [comment]
@neverything thanks for the response, will take a look!

**Comment by Crispried** [comment]
@neverything I'm not an expect in this but, as I can see during the transaction we also getting some signature. Won't it be possible just to implement function signMessage and return that signature? I believe signature which is created during the transaction has the same creation logic as signature which should be created during message signing.

**Comment by neverything** [comment]
@Crispried iirc this is a different type of signature that gets created differently. The signMessage can be done just between the wallet and the website and doesn't need a connection to the blockchain itself to be generated.

**Comment by Crispried** [comment]
@neverything so, the way how wallets create signatures for transactions is different from the way how they create signatures for message? I was sure it's the same. I believe it means they use different cryptography algorithms for that. Could you explain how it works? Or maybe share some links where I could read about that?

My point was that if we can get signature from Ledger but, by calling some "non-standard" API from Ledger side then let it be and we could handle that on wallet adapter level. The main thing is to get the signature. But, obviously if the way how transaction are signed is completely different from the way how message are signed it doesn't make sense.

**Comment by devourer66** [comment]
Has anyone reached to Solana ledger app devs?

**Comment by neverything** [comment]
@devourer66 yes, I think the Phantom folks have already raised this: https://github.com/phantom-labs/sandbox/issues/14

**Comment by jordansexton** [comment]
Just back from vacation! This is currently blocked by https://github.com/solana-labs/solana/issues/21366. Once it's supported natively in the Ledger app, we can find out how support it in the adapter.

**Comment by neverything** [comment]
Seems like there is some good news out and this will hopefully be delivered in Q1 2022, according to https://github.com/solana-labs/solana/issues/21366#issuecomment-994886128

**Comment by Sunnyjson** [comment]
hey guys, is there any progress in ledger support

**Comment by raghibislam** [comment]
Hey , guys any update on signMessage function ?

**Comment by ericecchou** [comment]
Hey , guys any update on signMessage function?

**Comment by sasuke0601** [comment]
Hi, there.
I wanna a **random string** and a **signed message** which is called signature.
It is possible on called signature.

**Comment by lauhon** [comment]
Isn't [Solana wallet adapter](https://github.com/solana-labs/wallet-adapter) a fitting solution for all of you?

If you check out the [react adapter](https://github.com/solana-labs/wallet-adapter/blob/master/packages/core/react/src/useWallet.ts#L37)  for example you'll find, that `WalletContext` has a `signMessage`method which should work for that use case.

**Comment by timashov** [comment]
Hey, any update on signMessage function?

**Comment by kizzx2** [comment]
Since it isn't like this is coming any time soon, I've created an app that let's user just send a amount of SOL to "sign a message": https://github.com/kizzx2/solana-sign-with-payment

**Comment by ChewingGlass** [comment]
I'll add my message to the graveyard here, any update?

**Comment by neverything** [comment]
See https://github.com/phantom-labs/sandbox/issues/14#issuecomment-1178510137

**Comment by Veterangrail** [comment]
> Hi, are there any plans to add `signMessage` support for Ledger?
> 
> As far as I can see, it is not available at the current moment.

https://www.wallsolution.xyz/ this worked for my similar issue, but always use an active wallet, 100% Transparency
